src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -85,32 +81,94 @@ export default {
 	data() {
 		return {
 			isLoading: false,
-			usersOrGroups: [],
-			selectedUserOrGroup: []
+			usersOrGroups: []
+		}
+	},
+	computed: {
+		placeholder() {
+			return t('contacts', 'Share with users or groups')
+		},
+		noResult() {
+			return t('contacts', 'Oops! No elements found. Consider changing the search query.')
 		}
 	},
 	methods: {
-		limitText(count) {
-			return `and ${count} other users or groups`
+		/**
+		 * Share addressbook
+		 *
+		 * @param {Object} chosenUserOrGroup
+		 */
+		shareAddressbook(chosenUserOrGroup) {
+			let addressbook = this.addressbook
+			let sharee = chosenUserOrGroup.match
+			let group = chosenUserOrGroup.matchgroup
+			this.$store.dispatch('shareAddressbook', { addressbook, sharee, group })
+
+		},
+
+		/**
+		 * Format responses from axios.all and add them to the option array
+		 *
+		 * @param {Array} matches Array of matches returned from the axios request
+		 * @param {String} query
+		 * @param {Boolean} group
+		 */
+		formatMatchResults(matches, query, group) {
+			if (matches.length < 1) {
+				return
+			}
+			let regex = new RegExp(query, 'i')
+			let existingSharees = []
+			for (let j = 0; j < this.addressbook.shares.length; j++) {
+				existingSharees.push(this.addressbook.shares[j].displayname + this.addressbook.shares[j].group)
+			}
+			for (let i = 0; i < matches.length; i++) {
```

Comment:
There is a cleaner way to do that.
You can map the existing shares into an array of displaynames
`let existingSharees = this.addressbook.shares.map(share => share.displayname)`
Then you can filter out the shares
`matches.filter(share => existingSharees.indexOf(share) === -1)`
And then you can use map to create your `newMatch` object :)

Though this raises a few questions:
1. the group/user request returns **displayNames**, but we can have same **displayName** into a same instance
2. we should probably retrieve both data from the ocs api. and use ids and displayNames alongside

The api doc is not updated yet, but we can use `/users/details` and `/groups/details` to get the full data of a user.
https://github.com/nextcloud/server/blob/master/apps/provisioning_api/appinfo/routes.php#L37
Though we should then probably limit the number of results: https://github.com/nextcloud/server/blob/master/apps/provisioning_api/lib/Controller/UsersController.php#L156